fs/src/io_uring/memory.rs

```diff
@@ -14,64 +15,67 @@ use {
 // and chunk it in slices of up to 1G each.
 const FIXED_BUFFER_LEN: usize = 1024 * 1024 * 1024;
 
-pub enum LargeBuffer {
-    Vec(Vec<u8>),
-    HugeTable(PageAlignedMemory),
-}
-
-impl Deref for LargeBuffer {
-    type Target = [u8];
+#[derive(Debug)]
+pub struct AllocError;
```

Comment:
the typical way to define errors that encode several conditions is by adding `thiserror` dependency and defining error as
```rust
use thiserror::Error;

#[derive(Debug, Error)]
pub enum AllocError {
   #[error("message")]
   SomeCondition,
   #[error("failed due to {0}")]
   Wrapping(#[from] AllocError)
}
```

see https://docs.rs/thiserror/latest/thiserror/

I think we can use it here